lib/std/heap.zig

```diff
@@ -38,37 +38,63 @@ const CAllocator = struct {
             pub const supports_malloc_size = true;
             pub const malloc_size = c.malloc_usable_size;
         }
+    else if (comptime @hasDecl(c, "_msize"))
+        struct {
+            pub const supports_malloc_size = true;
+            pub const malloc_size = c._msize;
+        }
     else
         struct {
             pub const supports_malloc_size = false;
         };
 
-    // The alignment guaranteed by malloc, the value matches the result of the C
-    // expression `alignof(max_alignment_t)`
-    const min_ptr_alignment = comptime std.math.max(
-        @alignOf(c_longdouble),
-        @alignOf(c_longlong),
-    );
+    pub const supports_posix_memalign = false and @hasDecl(c, "posix_memalign");
 
-    fn aligned_alloc(len: usize, alignment: usize) ?*c_void {
-        // The minimum alignment supported by both APIs is the size of a pointer
-        const eff_alignment = std.math.max(alignment, @sizeOf(usize));
+    fn get_header(ptr: [*]u8) *[*]u8 {
+        return @intToPtr(*[*]u8, @ptrToInt(ptr) - @sizeOf(usize));
+    }
 
-        if (builtin.os.tag == .windows) {
-            return c._aligned_malloc(len, eff_alignment);
+    fn aligned_alloc(len: usize, alignment: usize) ?[*]u8 {
```

Comment:
this could have return type `?[*]align(@alignOf(usize)) u8`

Reply:
The alternative code path may not have the same minimum alignment.